drivers/serial/uart_ra.c

```diff
@@ -366,28 +673,68 @@ static const struct uart_driver_api uart_ra_driver_api = {
+				   DT_IRQ_BY_NAME(DT_INST_PARENT(n), name, priority), isr, dev,    \
+				   DT_IRQ_BY_NAME(DT_INST_PARENT(n), name, flags));
+
+#define RA_IRQ_DISCONNECT_DYNAMIC(n, name, dev, isr)                                               \
+	ra_icu_irq_disconnect_dynamic(irqn, 0, NULL, NULL, 0)
+
+#define UART_RA_CONFIG_FUNC(n)                                                                     \
+	static int irq_config_func_##n(const struct device *dev)                                   \
+	{                                                                                          \
+		struct uart_ra_data *data = dev->data;                                             \
+		int irqn;                                                                          \
+                                                                                                   \
+		irqn = RA_IRQ_CONNECT_DYNAMIC(n, rxi, dev, uart_ra_isr_rxi);                       \
+		if (irqn < 0) {                                                                    \
+			return irqn;                                                               \
+		}                                                                                  \
+		data->irqn[UART_RA_INT_RXI] = irqn;                                                \
+		irqn = RA_IRQ_CONNECT_DYNAMIC(n, txi, dev, uart_ra_isr_txi);                       \
+		if (irqn < 0) {                                                                    \
+			goto err_txi;                                                              \
+		}                                                                                  \
+		data->irqn[UART_RA_INT_TXI] = irqn;                                                \
+		irqn = RA_IRQ_CONNECT_DYNAMIC(n, eri, dev, uart_ra_isr_eri);                       \
+		if (irqn < 0) {                                                                    \
+			goto err_eri;                                                              \
+		}                                                                                  \
+		data->irqn[UART_RA_INT_ERI] = irqn;                                                \
+		return 0;                                                                          \
+                                                                                                   \
+err_eri:                                                                                           \
+		RA_IRQ_DISCONNECT_DYNAMIC(data->irq[UART_RA_INT_TXI], eri, dev, uart_ra_isr_eri);  \
+err_txi:                                                                                           \
+		RA_IRQ_DISCONNECT_DYNAMIC(data->irq[UART_RA_INT_RXI], txi, dev, uart_ra_isr_txi);  \
+                                                                                                   \
+		return irqn;                                                                       \
```

Comment:
why did you decide to implement this as a macro? I can't see anything hardcoded in the body, just make it a normal function, if the compiler complains in some config slap a `__maybe_unused` on it